secretsmanager/include/secrets_manager_enc.h

```diff
@@ -27,34 +27,31 @@
 
 namespace concord::secretsmanager {
 
+struct KeyParams;
+class AES_CBC;
+
 // SecretsManagerEnc handles encryption and decryption of files.
 // The following flow is used for encryption:
-// 1. A random salt is generated
-// 2. The salt and the password (provided by the configuration) are used to derive a secret key and iv. OpenSSL
-//   OPENSSL_EVP_BytesToKey() is used for key derivation (openssl_pass.h).
-// 3. The key is used to encrypt the payload with a supported block cipher (AES256 CBC).
-// 4. The encrypted cipher text and the salt are concatenated.
-//    The contents of the buffer is: Salted__<8 bytes salt><cipher>
-// 5. The buffer is base64 encoded and written to file.
+// 1. SecretData are the input parameters for SecretsManagerEnc. They contain an algorithm name, symmetric key and IV.
+// 2. The key and iv are used to encrypt the payload with a supported block cipher (AES256 CBC).
+// 3. The buffer is base64 encoded and written to file.
 class SecretsManagerEnc : public ISecretsManagerImpl {
-  std::string password_;
-  std::string digest_;
-  std::string algo_;
-  uint32_t key_length_;
-
   const std::set<std::string> supported_encs_{"AES/CBC/PKCS5Padding", "AES/CBC/PKCS7Padding"};
-  const std::set<std::string> supported_digests_{"SHA-256"};
-  const uint32_t SALT_SIZE = 8;
 
-  logging::Logger logger = logging::getLogger("concord.bft.secrets-manager-enc");
-  CryptoPP::BlockingRng rand;
+  logging::Logger logger_ = logging::getLogger("concord.bft.secrets-manager-enc");
```

Comment:
Nit: The files in our code base, by convention,  almost always put the private data fields below the public methods with an explicit `private:` section.